src/components/validators/ValidatorData.vue

```diff
@@ -0,0 +1,217 @@
+    const currentVote = computed(() => store.state.account.vote);
+    const showCpu = ref<boolean>(false);
+    const voteChanged = ref<boolean>(false);
+    const resetFlag = ref<boolean>(false);
+    const lastWeight = ref<string>('0');
+    const lastStaked = ref<number>(0);
+    const stakedAmount = ref<number>(0);
+    const accountValid = computed(() => account.value && account.value !== '');
+    const transactionId = ref<string>(store.state.account.TransactionId);
+    const transactionError = ref<unknown>(store.state.account.TransactionError);
+    const openTransaction = ref<boolean>(false);
+    const payrate = ref(0);
+    const top21pay24h = ref(0);
+    const supply = ref(0);
+    const voters = ref(0);
+    const amount_voted = ref(0);
+    const votesProgress = computed(() => {
+      return amount_voted.value / supply.value || 0;
+    });
+
+    async function getVotes() {
+      if (account.value && account.value !== '') {
+        const data = await api.getAccount(account.value);
+        store.commit('account/setAccountData', data);
+        const voterInfo = data.account.voter_info;
+        producerVotes.value = voterInfo.producers;
+        lastWeight.value = parseFloat(voterInfo.last_vote_weight).toFixed(2);
+        lastStaked.value = voterInfo.last_stake;
+        stakedAmount.value = voterInfo.staked;
+      }
+    }
+    function assetToAmount(asset: string, decimals = -1): number {
+      try {
+        let qty: string = asset.split(' ')[0];
+        let val: number = parseFloat(qty);
+        if (decimals > -1) qty = val.toFixed(decimals);
+        return val;
+      } catch (error) {
+        return 0;
+      }
+    }
+    async function getVotingStatistics() {
+      const request = {
+        code: 'eosio',
+        lower_bound: 'eosio',
+        table: 'voters'
+      };
+      voters.value = (await api.getTableByScope(request))[0].count;
+      const totalStakeParams = {
+        code: 'eosio',
+        scope: 'eosio',
+        table: 'global'
+      } as GetTableRowsParams;
+      amount_voted.value =
+        (
+          (await api.getTableRows(totalStakeParams)) as {
+            rows: { total_activated_stake: number }[];
+          }
+        ).rows[0].total_activated_stake / 10000;
+
+      const paramsSupply = {
+        code: 'eosio.token',
+        scope: 'TLOS',
+        table: 'stat'
+      } as GetTableRowsParams;
+      supply.value = assetToAmount(
+        (
+          (await api.getTableRows(paramsSupply)) as {
+            rows: { supply: string }[];
+          }
+        ).rows[0].supply
+      );
+      const paramsPayrate = {
+        code: 'eosio',
+        scope: 'eosio',
+        table: 'payrate'
+      } as GetTableRowsParams;
+      payrate.value = (
+        (await api.getTableRows(paramsPayrate)) as {
+          rows: { bpay_rate: number }[];
+        }
+      ).rows[0].bpay_rate;
+      top21pay24h.value = ((payrate.value / 100000) * supply.value) / 365 / 21;
+    }
```

Comment:
I would refactor these into separate methods `getPayRate`, `getSupply`, `getStake`, etc if they don't depend on variables used else where in the parent method

Reply:
Okay cool will make the changes thanks @DonaldPeat

Reply:
![image](https://user-images.githubusercontent.com/41372145/176435412-81bceb3a-ab3b-4678-a860-ee725308f067.png)

Is this fine @DonaldPeat